assets/js/search-categories-metabox.js

```diff
@@ -0,0 +1,186 @@
+/* global selectedCategories:true */
+
+(function( $ ) {
+
+	$( 'document' ).ready(function() {
+
+		// Global declarations
+
+		// Contains selected Category ID to be sent in AJAX request.
+		var selectedCategories = getSelectedCategories();
+
+		// Function definitions.
+
+		/**
+		 * @summary Sorts the Categories by name and then by checked items.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @global type selectedCategories List of selected Categories ID.
+		 */
+		function sortCategoriesListBySelectedItems() {
+
+			var selectedCategories = [],
+				otherCategories    = [],
+				checkBoxState;
+
+			$( 'ul#scm-categories-ul li' ).each(function () {
+				checkBoxState = $( this ).find( 'input' ).prop( 'checked' );
+				if ( true === checkBoxState ) {
+					selectedCategories.push( $( this ) );
+				} else {
+					otherCategories.push( $( this ) )
+				}
+			});
+
+			$( 'ul#scm-categories-ul' )
+				.html( selectedCategories )
+				.append( otherCategories );
+		}
+
+		/**
+		 * @summary Gets the User selected Categories.
+		 *
+		 * @since 0.1.0
+		 *
+		 * @return array List of User selected Categories.
+		 */
+		function getSelectedCategories() {
+			// Category ID's.
```

Comment:
It is not a comment, but just a suggestion. Also it is just my personal preference/opinion.

I feel that you are over commenting the code. Generally comments should just say why and not what or how. What or how should be clear enough if the code id readable.

The problem with over commenting is that, it is very easy for it to get outdated when you make code changes. Also for someone who can read code it is redundant.

Eg:

```
// Return the array of selected Category ID's.
return categories;
```

It is very obvious what you are returning.